ME2Randomizer/Classes/Randomizers/ME2/Misc/RTexts.cs

```diff
@@ -191,34 +193,37 @@ private static string AddReactionToLine(string vanillaLine, string modifiedLine,
-            if (modifiedLine.Length < 2 || Regex.Matches(modifiedLine, regexAllLetters).Count == 0 || vanillaLine.Contains('{'))
+            //initialize reactions/regex if this is first run
+            if (ReactionList == null)
             {
-                //I should go.
-                return modifiedLine;
+                string rawReactionDefinitions = MERUtilities.GetEmbeddedStaticFilesTextFile("reactiondefinitions.xml");
+                StringReader reactionReader = new StringReader(rawReactionDefinitions);
+                XmlSerializer serializer = new XmlSerializer(typeof(List<Reaction>), new XmlRootAttribute("ReactionDefinitions"));
+                ReactionList = (List<Reaction>)serializer.Deserialize(reactionReader);
+
+                reactionRegexDictionary.Add("EndOfSentence", new Regex(@"(?<![M| |n][M|D|r][s|r]\.)(?<!(,""))(?<=[.!?""])(?= [A-Z])", RegexOptions.Compiled));
+                reactionRegexDictionary.Add("AllLetters", new Regex("[a-zA-Z]", RegexOptions.Compiled));
+                reactionRegexDictionary.Add("PunctuationRemover", new Regex("(?<![D|M|r][w|r|s])[.!?](?!.)", RegexOptions.Compiled));
+                reactionRegexDictionary.Add("BorkedElipsesFixer", new Regex("(?<!\\.)\\.\\.(?=\\s|$)", RegexOptions.Compiled));
             }
 
-            char[] reactionDangerousCharacters = new char[] { '<', '\n' };
-            if (modifiedLine.IndexOfAny(reactionDangerousCharacters) >= 0 || vanillaLine.Length > 200)
+            if (modifiedLine.Length < 2 || reactionRegexDictionary["AllLetters"].Matches(modifiedLine).Count == 0 || vanillaLine.Contains('{'))
             {
-                dangerousLine = true;
+                //I should go.
+                return modifiedLine;
             }
 
-            //initialize reactions if they aren't loaded
-            if (ReactionList == null)
+            char[] dangerousCharacters = { '<', '\n' };
+            if (modifiedLine.IndexOfAny(dangerousCharacters) >= 0 || vanillaLine.Length > 200)
             {
-                string rawReactionDefinitions = MERUtilities.GetEmbeddedStaticFilesTextFile("reactiondefinitions.xml");
-                StringReader reactionReader = new StringReader(rawReactionDefinitions);
-                XmlSerializer serializer = new XmlSerializer(typeof(List<Reaction>), new XmlRootAttribute("ReactionDefinitions"));
-                ReactionList = (List<Reaction>)serializer.Deserialize(reactionReader);
+                dangerousLine = true;
             }
 
             //split strings into sentences for processing
-            string pattern = @"(?<![M| |n][M|D|r][s|r]\.)(?<!(,""))(?<=[.!?""])(?= [A-Z])";
             List<string> splitVanilla = new List<string>();
             List<string> splitModified = new List<string>();
 
-            MatchCollection regexMatches = Regex.Matches(vanillaLine, pattern);
+            MatchCollection regexMatches = reactionRegexDictionary["EndOfSentence"].Matches(vanillaLine);
```

Comment:
One thing that may improve speed here is to just hardcode the EndOfSentence regex variable. This way a lookup doesn't need to be performed at all - since this is not a dynamic item.

Reply:
e.g. initialize this regex once and then just directly call endOfSentenceRegex.Matches(vanillaLine) and pass that regex around

Reply:
Actually had it this way initially, but didn't like how much I was declaring outside of the function. Went back and just got rid of the dictionary and hardcoded the common regex.